### DISCUSSION TITLE
how to get access token from https://microsoft.com/devicelogin with curl?

### DISCUSSION THREAD

**Discussion by pheiduck** [curl/curl#8882] [Q&A] [answered]
Hey Curl Community and Team,

I need to automate getting tokens from https://microsoft.com/devicelogin is there a way to do this with curl if yes, how?

Thanks for your help in advance.

Kind Regards,
Philip H.

**Comment by jay** [comment, marked as answer]
> I need to automate getting tokens from https://microsoft.com/devicelogin is there a way to do this with curl if yes, how?

If there's no public API for it you'll need to use a [debugging proxy](https://www.telerik.com/fiddler) but that's outside the scope of what we can help you with.
